include/eve/module/real/core/function/regular/simd/x86/compress_store.hpp

```diff
@@ -59,9 +118,27 @@ namespace eve::detail
                      wide<T, N> v,
                      logical<wide<T, N>> mask,
                      Ptr ptr) noexcept
-    requires std::same_as<abi_t<T, N>, x86_128_> && (N() == 4)
+    requires x86_abi<abi_t<T, N>> && ( N() == 4 )
   {
-    if constexpr ( std::floating_point<T> )
+    if constexpr ( N() == 4 && sizeof(T) == 8 && current_api == avx  )
+    {
+      return compress_store_aggregated_unsafe(v, mask, ptr);
+    }
+    else if constexpr ( N() == 4 && sizeof(T) == 8 )
+    {
+      alignas(32) auto patterns = pattern_4_elements(idx_dwords<std::uint64_t>);
```

Comment:
why is htis 32 hardcoded ?

Reply:
`if constexpr ( N() == 4 && sizeof(T) == 8 )`

Reply:
Will do in the next patch.